src/lib.rs

```diff
@@ -75,11 +75,17 @@ impl Bibliography {
 
     /// Parse a bibliography from a source string.
     pub fn parse(src: &str) -> Option<Self> {
+        Self::from_raw(RawBibliography::parse(src)).ok()
+    }
+
+    /// Same as parse but return a Result instead of an Option. The Err variant contains the key
+    /// of the entry that failed to parse
+    pub fn parse_result(src: &str) -> Result<Self, String> {
         Self::from_raw(RawBibliography::parse(src))
     }
```

Comment:
Your new `parse_result` API should be the default since it is not to hard for the caller to do `.ok()` themselves if they really want the old behavior.

It _would_ be a breaking API change but that's okay I think because BibLaTeX is <1.0 anyways.

So I'd propose merge this into a single method:
```rust
pub fn parse(src: &str) -> Result<Self, String>
```

Reply:
Thanks for your kind comments. 

The reason I did this in a separate function was to make the changes non-beaking. But since it is ok to make breaking changes, I will make the standard parse function return a Result, as you suggest above.

I've also incorporated all your other suggestions.

Before pushing again, one additional change I thought about would be to change this part of `from_raw`:

```
            let fields = entry
                .fields
                .into_iter()
                .filter_map(|(key, value)| {
                    if let Some(r) = resolve::resolve(value, abbr) {
                        Some((key.to_string(), r))
                    } else {
                        None
                    }
                })
                .collect::<HashMap<String, Chunks>>();
            if fields.len() != count {
                return Err(entry.key.to_string());
            }
```

Instead of using `None` and `filter_map`, we could also register the names of the fields that are causing trouble. In order to not make the API too complicated, I could replace the above with:

```
            let mut fields = HashMap::new();
            for (field_key, field_value) in entry.fields.into_iter() {
                if let Some(r) = resolve::resolve(field_value, abbr) {
                    fields.insert(field_key.to_string(), r);
                } else {
                    // Instead of continuing to parse on an incorrect field, return right away
                    return Err((entry.key.to_string(), field_key.to_string()));
                }
            }
```
(if I understand the code correctly, this will just return on the first error field it encounters, there could be others).
The signature of `parse` would then be `pub fn parse(src: &str) -> Result<Self, (String, String)>`

This seems to work in my tests. Do you think it is worth the extra complexity?